### PR TITLE
list-plugins: improve script

### PR DESCRIPTION
**Pull request by GaetanLepage**
- **dev/list-plugins: fix spacing in plugin printing (deprecation warning)**
- **dev/list-plugins: add symbol for 'MISC' type plugins**
- **dev/list-plugins: update EXCLUDES and KNOWN_PATHS variables**

### CODE REVIEW

**Comment by khaneliman** [comment]
Ooh nice, gonna test real quick. I've been using this for the tracking issue and just general usage for a while.

**Comment by khaneliman** [comment]
Does look like unknown is shifted atm 
image
Don't know if that's like that on main

File: flake-modules/dev/list-plugins.py [hunk 1]
```diff
@@ -11,10 +11,38 @@
 
 EXCLUDES: list[str] = [
     # Not plugin files
```
Comment (MattSturgeon) [inline flake-modules/dev/list-plugins.py:13]: Maybe in the future we can detect plugins files? Or maybe that's non-trivial...

We could use the same strategy as the "by-name" and "package option" tests, i.e. scan the toplevel option set recursively, collect all options, then filter for `enable` options.

The option itself contains info on where it was declared, so we get the filename that way.

Obviously beyond this PR, and would need some nix code to complement the python.

File: flake-modules/dev/list-plugins.py [hunk 2]
```diff
@@ -31,10 +59,71 @@ class State(Enum):
...
+        bool,  # Has deprecation warnings
+    ],
+] = {
+    "plugins/by-name/chadtree/default.nix": (State.OLD, Kind.NEOVIM, False),
+    "plugins/by-name/commentary/default.nix": (State.OLD, Kind.VIM, False),
+    "plugins/by-name/conjure/default.nix": (
+        State.OLD,
+        Kind.VIM,  # configured with weird globals `vim.g["conjure#mapping#doc_word"] = "gk"`
+        False,
+    ),
```
Comment (MattSturgeon) [inline flake-modules/dev/list-plugins.py:76]: Doesn't having this hard-coded defeat the purpose of this test? Can you try and explain why we're doing things this way?

**Comment by khaneliman** [comment]
Latest commit seems to have gotten rid of those question marks, now the only weird output I see 
image

File: flake-modules/dev/list-plugins.py (hunk 2 above)
Comment (GaetanLepage) [reply flake-modules/dev/list-plugins.py:75]: Well, for me, having a plugin listed as "unknown" is not helping at all.
So we have two choices:
1. improve the matching logic to automatically identify more plugins
2. Add exceptions

But anyway, I pushed a commit that completely get rid of `kind.UNKNOWN`.

**Comment by GaetanLepage** [comment]
> Latest commit seems to have gotten rid of those question marks, now the only weird output I see

I have fixed it so that it looks good in my terminal. But the issue might be on my end, so I will revert this.

**Comment by khaneliman** [comment]
> > Latest commit seems to have gotten rid of those question marks, now the only weird output I see
> 
> I have fixed it so that it looks good in my terminal. But the issue might be on my end, so I will revert this.

Oh, maybe we can't handle all different terminal/font/etc combinations.

**Review by khaneliman — APPROVED** [review]
LGTM, thanks for updated functionality!

The amount of manual workarounds indicates we might benefit from some more meta attributes in our modules that could help identify plugins better.

**Comment by GaetanLepage** [comment]
> The amount of manual workarounds indicates we might benefit from some more meta attributes in our modules that could help identify plugins better.

Honestly, I hope for a bug push to convert all plugins and deprecate this script haha !

**Comment by GaetanLepage** [comment]
@mergifyio queue

**Comment by khaneliman** [comment]
> > The amount of manual workarounds indicates we might benefit from some more meta attributes in our modules that could help identify plugins better.
> 
> Honestly, I hope for a bug push to convert all plugins and deprecate this script haha !

I could still see a usage for it for querying our plugins easily and allow searching / filtering them by neovim/vim plugin/colorscheme etc

**Comment by GaetanLepage** [comment]
> I could still see a usage for it for querying our plugins easily and allow searching / filtering them by neovim/vim plugin/colorscheme etc

For sure.

**Pull request merged**